### DISCUSSION TITLE
Declaring new Property in aircraft .xml

### DISCUSSION THREAD

**Discussion by ThomasKennings** [JSBSim-Team/jsbsim#661] [Q&A]
Documentation like the JSBSim PDF manual suggests that it's totally possible to instantiate an aircraft-specific variable, but when I try to do so I get an invalid token error at the bottom of the 737.xml file. I am trying to make an altitude setpoint variable for use in a PID flight controller. I have tried several things, among them putting this next to my PID section:

<property value="10000"> fcs/pid-alt-setpoint </property>

Can someone give me an example of the proper way to do this? For the life of me I cannot find where other aircraft-specific properties are declared. I apologize for clogging this forum so!

**Comment by seanmcleod** [comment]
Please post the actual snippet and also post the actual output error. It makes it much more difficult and partly a guessing game trying to debug issues with very limited info.

Here are 2 random examples. A good source is to look for existing examples in the repo.

https://github.com/JSBSim-Team/jsbsim/blob/33735db0fb1e4fdf47f253a4215dbe3bb63a8e92/aircraft/global5000/global5000ap.xml#L30-L53

https://github.com/JSBSim-Team/jsbsim/blob/33735db0fb1e4fdf47f253a4215dbe3bb63a8e92/aircraft/global5000/global5000.xml#L250-L253
